### PR TITLE
Feature: Test server identification too long

### PR DESCRIPTION
**Pull request by lwojtow**
* Chore: Rename `user_agent` config field to `ident`
* Feature: Add test for large 'Server' header during handshake.

### CODE REVIEW

File: src/tests/resistance/handshake.rs [hunk 1]
```diff
@@ -38,3 +42,42 @@ async fn r001_HANDSHAKE_reject_if_user_agent_too_long() {
     synth_node2.shut_down().await;
     node.stop().unwrap();
 }
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r002_HANDSHAKE_reject_if_server_too_long() {
```
Comment (Rqnsom) [inline src/tests/resistance/handshake.rs:48]: Maybe these two tests could be placed under the same SPEC test R001 like:
r001_t1_HANDSHAKE_xxx
r001_t2_HANDSHAKE_xxx

So we have different variants of the test: "test with wrong handshake fields"

Comment (Rqnsom) [reply src/tests/resistance/handshake.rs:48]: Because even the SPEC texts are pretty similar, they both revolve around doing an invalid handshake thing.

**Review by Rqnsom — APPROVED** [review]
LGTM 👍

File: src/tests/resistance/handshake.rs (hunk 1 above)
Comment (lwojtow) [reply src/tests/resistance/handshake.rs:48]: Done

**Pull request merged**